src/components/vm/confirmDialog.jsx

```diff
@@ -0,0 +1,131 @@
+import { Button } from "@patternfly/react-core/dist/esm/components/Button";
+import { DescriptionList, DescriptionListDescription, DescriptionListGroup, DescriptionListTerm } from "@patternfly/react-core/dist/esm/components/DescriptionList";
+import { Spinner } from "@patternfly/react-core/dist/esm/components/Spinner";
+import { Modal } from "@patternfly/react-core/dist/esm/components/Modal";
+import { useDialogs } from 'dialogs.jsx';
+import { distanceToNow } from 'timeformat.js';
+
+import { domainGetStartTime } from '../../libvirtApi/domain.js';
+import { Enum } from '../../libvirtApi/helpers.js';
+
+const _ = cockpit.gettext;
+
+export const ConfirmDialog = ({ idPrefix, actionsList, title, titleIcon, vm }) => {
+    const Dialogs = useDialogs();
+    const [uptime, setUptime] = useState();
+    const [dialogLoading, setDialogLoading] = useState(true);
+
+    useEffect(() => {
+        return domainGetStartTime({ connectionName: vm.connectionName, vmName: vm.name })
+                .then(res => {
+                    setUptime(res);
+                    setDialogLoading(false);
+                });
+    }, [vm]);
+
+    const actions = actionsList.map(action =>
+        <Button variant={action.variant}
+            key={action.id}
+            id={`${idPrefix}-${action.id}`}
+            onClick={() => {
+                action.handler();
+                Dialogs.close();
+            }}>
+            {action.name}
+        </Button>
+    );
+    actions.push(
+        <Button variant="link" key="cancel" onClick={Dialogs.close}>
+            {_("Cancel")}
+        </Button>
+    );
+
+    let body;
+    if (dialogLoading) {
+        body = (
+            <Bullseye>
+                <Spinner />
+            </Bullseye>
+        );
+    } else if (uptime) {
+        let uptimeStr = distanceToNow(new Date(uptime));
+
+        // https://libvirt.org/html/libvirt-libvirt-domain.html#virDomainRunningReason
+        switch (vm.stateReason) {
+        case Enum.VIR_DOMAIN_RUNNING_MIGRATED:
+        case Enum.VIR_DOMAIN_RUNNING_POSTCOPY: {
+            uptimeStr = cockpit.format(_("$0 since migration"), uptimeStr);
+            break;
+        }
+        case Enum.VIR_DOMAIN_RUNNING_FROM_SNAPSHOT: {
+            uptimeStr = cockpit.format(_("$0 since reverting to snapshot"), uptimeStr);
+            break;
+        }
+        case Enum.VIR_DOMAIN_RUNNING_RESTORED:
+        case Enum.VIR_DOMAIN_RUNNING_UNPAUSED: {
+            uptimeStr = cockpit.format(_("$0 since VM was resumed"), uptimeStr);
+            break;
+        }
+        case Enum.VIR_DOMAIN_RUNNING_MIGRATION_CANCELED:
+        case Enum.VIR_DOMAIN_RUNNING_POSTCOPY_FAILED: {
+            uptimeStr = cockpit.format(_("$0 since migration was cancelled"), uptimeStr);
+            break;
+        }
+        case Enum.VIR_DOMAIN_RUNNING_SAVE_CANCELED: {
+            uptimeStr = cockpit.format(_("$0 since failed save process"), uptimeStr);
+            break;
```

Comment:
These 3 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1069-20230621-115207-690ab5dc-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)